### PR TITLE
Porting to erigon 2.28

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkrpc/commands/eth_api.cpp [hunk 1]
```diff
@@ -157,7 +157,13 @@ boost::asio::awaitable<void> EthereumRpcApi::handle_eth_gas_price(const nlohmann
...
 
         GasPriceOracle gas_price_oracle{ block_provider};
-        const auto gas_price = co_await gas_price_oracle.suggested_price(block_number);
+        auto gas_price = co_await gas_price_oracle.suggested_price(block_number);
+
+        const auto block_with_hash = co_await block_provider(block_number);
+        const auto &base_fee = block_with_hash.block.header.base_fee_per_gas.value_or(0);
+        if (base_fee != 0) {
+           gas_price += base_fee;
+        }
```
Comment (canepat) [inline silkrpc/commands/eth_api.cpp:166]: Here I would suggest one of the following approaches to improve readability:

1)
```
const auto base_fee = block_with_hash.block.header.base_fee_per_gas;
if (base_fee) {
    gas_price += base_fee;
}
```

2)
```
const auto base_fee = block_with_hash.block.header.base_fee_per_gas.value_or(0);
gas_price += base_fee;
```

File: silkrpc/core/gas_price_oracle.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ boost::asio::awaitable<void> GasPriceOracle::load_block_prices(uint64_t block_nu
     int idx = 0;
     block_prices.reserve(block_with_hash.block.transactions.size());
     for (const auto& transaction : block_with_hash.block.transactions) {
-        const auto effective_gas_price = transaction.effective_gas_price(base_fee);
+        const auto effective_gas_price = transaction.priority_fee_per_gas(base_fee);
```
Comment (canepat) [inline silkrpc/core/gas_price_oracle.cpp:84]: `effective_gas_price` -> `priority_fee_per_gas` starting from here

File: silkrpc/core/gas_price_oracle_test.cpp [hunk 3]
```diff
@@ -343,7 +343,7 @@ TEST_CASE("suggested price") {
     }
 
     SECTION("when there are 60 blocks with with 0x07 base fee with fee == kDefaultMaxPrice") {
-        FixedBlockData data = {0x07, kDefaultMaxPrice, kDefaultMaxPrice, kDefaultMaxPrice, kDefaultMaxPrice};
+        FixedBlockData data = {0x07, kDefaultMaxPrice+0x07, kDefaultMaxPrice+0x07, kDefaultMaxPrice+0x07, kDefaultMaxPrice+0x07};
```
Comment (canepat) [inline silkrpc/core/gas_price_oracle_test.cpp:346]: `kDefaultMaxPrice+0x07` -> `kDefaultMaxPrice + 0x07` to keep code more uniform

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkrpc/commands/eth_api.cpp (hunk 1 above)
Comment (lupin012) [reply silkrpc/commands/eth_api.cpp:166]: I have implemented option1

File: silkrpc/core/gas_price_oracle.cpp (hunk 2 above)
Comment (lupin012) [reply silkrpc/core/gas_price_oracle.cpp:84]: ok

File: silkrpc/core/gas_price_oracle_test.cpp (hunk 3 above)
Comment (lupin012) [reply silkrpc/core/gas_price_oracle_test.cpp:346]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**